### DISCUSSION TITLE
Videos

### DISCUSSION THREAD

**Discussion by SHadley1138** [RhetTbull/osxphotos#1261] [Q&A]
Hi there,

love this code!!  but it seems to be skipping quite a few videos, and I can't seem to see why.

osxphotos export ~/Documents/PhotoBackup --skip-original-if-edited --exiftool --directory "{folder_album}" --retry 3 —force-update --report ~/Documents/PhotoBackup/export.csv --verbose --convert-to-jpeg --incloud  --shared-library --deleted --update-errors

has anyone else ever seen this?

**Comment by RhetTbull** [comment]
Find a video in the report and look to see if the missing column is True (1 in the column) for that video. 

Do you have optimize mac storage set in Photos Settings > iCloud? 

If you can find a video it skips, run your command again with the addition of these flags:

`--verbose --name VIDEO_NAME.mov` where you replace `VIDEO_NAME.mov` with the name of the actual video that was skipped. What does the verbose out put print?

**Reply by SHadley1138** [reply]
thank for the reply.  I have optimised Mac storage enabled, as my library is about 1TB.  looking at the log file, it says it is missing, but I can play the video from the photo library, so I know it isn't missing in reality.  forcing an export of that file name, reveals there are two with the same name.  shouldn't be a big deal as they are in different folders/albums
[export.csv](https://github.com/RhetTbull/osxphotos/files/13125206/export.csv)

**Reply by RhetTbull** [reply]
If you have "Optimize Mac storage" enabled then the video is "missing" in the sense it's not cached on the local Mac but kept in iCloud. It plays when you select it because at that moment, Photos streams it from iCloud. osxphotos works by copying files out of the Photos library but in this case the file isn't there, it's in iCloud, and only a small preview is available on the local Mac. You can add the `--download-missing` or `--download-missing --use-photokit` options (I recommend the latter) to tell osxphotos to force Photos to download the file from iCloud if missing.

**Reply by SHadley1138** [reply]
Excellent, I’ll give that a crack and let you know.  Thanks for your help!
